sqlglot/parser.py

```diff
@@ -1362,9 +1362,18 @@ def extend_props(temp_props: t.Optional[exp.Properties]) -> None:
             # exp.Properties.Location.POST_SCHEMA ("schema" here is the UDF's type signature)
             extend_props(self._parse_properties())
 
-            self._match(TokenType.ALIAS)
-
-            if self._match(TokenType.COMMAND):
+            heredoc_matched = False
+            if self._match(TokenType.ALIAS):
+                if self._match_text_seq("$", advance=False):
+                    expression = self._parse_heredoc()
+                    heredoc_matched = True
+                elif self._match(TokenType.HEREDOC_STRING):
+                    expression = self.expression(exp.Heredoc, this=self._prev.text)
+                    heredoc_matched = True
```

Comment:
should we consolidate these two branches by handling it all in `_parse_heredoc` and setting `heredoc_matched` to true if we get back a non-`None` result? just trying to see if we can simplify it a bit